### ISSUE TITLE
Output is always null if the step id contains a hyphen (-) in a scaffolder template

### ISSUE THREAD

**Issue by MrBlaise** [backstage/backstage#9725]
## Expected Behavior

When creating a step in a Template with an id that contains a hyphen (`-`) for example `publish-code`, I should be able to reference its output
in the outputs session like `steps.publish-code.output.remoteUrl`

## Actual Behavior
When a step id contains a hyphen the output cannot be referenced it will always be null. (assuming the handlebar worker fails in this case)

## Steps to Reproduce

- Create a backstage app
- Clone the react template, and reference it in the app-config.yaml
- Rename the publish step's id to `publish-something`
- Update the reference in the output section `${{ steps.publish-something.output.remoteUrl }}`
- Create a react app and observe that the link is not rendered at bottom (the remoteUrl output returned null)

**Comment by benjdlambert** [comment]
i think this could be related to `beta3` templates, a similar comment was referenced from https://github.com/backstage/backstage/issues/7686#issuecomment-1046233235

**Comment by benjdlambert** [comment]
I think you could try `${{ steps['publish-something'].output.remoteUrl }}`

**Comment by MrBlaise** [comment]
@benjdlambert Hey, thanks that works! :) 

We wrote our custom github pull request step (due to this: https://github.com/backstage/backstage/issues/9574) and we have spent a considerable amount of time trying to figure out what went wrong and why the output is always null, so this caused a little bit of a headache 😅

**Comment by benjdlambert** [comment]
Yeah the output of that other thread is to update the docs in the right places so that it's made clear if you're migrating from the beta2 syntax or just creating new templates that `-` are gonna cause a headache
